node/libs/crypto/src/bn254/mod.rs

```diff
@@ -81,6 +81,24 @@ impl PartialEq for SecretKey {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct PublicKey(G2);
 
+impl PublicKey {
+    /// Checks if the public key is not the identity element and is in the correct subgroup. Verifying signatures
+    /// against public keys that are not valid is insecure.
+    pub fn is_valid(&self) -> bool {
+        // Check that the point is not the identity element.
+        if self.0.is_zero() {
+            return false;
+        }
+
+        // We multiply the point by the order and check if the result is the identity element.
+        // If it is, then the point is on the correct subgroup.
+        let order = Fr::char();
+        let mut p = self.0;
+        p.mul_assign(order);
+        p.is_zero()
+    }
+}
+
 impl Default for PublicKey {
     fn default() -> Self {
         PublicKey(G2::zero())
```

Comment:
It is problematic that we allow to construct an invalid public key. We should either remove the default constructor, or keep the is_valid/verify public and revert my suggestion of using `assert!` instead of returning an error.

Reply:
Ok, I've tried to remove the default constructor and it compiles just fine with minimal changes to AggregateSignature::verify.